spec/update/gemfile_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+RSpec.describe "bundle update" do
+  context "with --gemfile" do
+    it "finds the gemfile" do
+      gemfile bundled_app("NotGemfile"), <<-G
+        source "file://#{gem_repo1}"
+        gem 'rack'
+      G
+
+      bundle! :install, :gemfile => bundled_app("NotGemfile")
+      bundle! :update, :all => bundle_update_requires_all?, :gemfile => bundled_app("NotGemfile")
+
+      ENV["BUNDLE_GEMFILE"] = "NotGemfile"
```

Comment:
@colby-swandale this is taken from the `bundle install --gemfile` tests too -- turns out that needs to be set so that the spec helper `the_bundle` retrieves the correct Gemfile. Updating both specs with a comment about this.